modules/core/src/norm.simd.hpp

```diff
@@ -0,0 +1,703 @@
+            r1 = v_add(r1, v_cvt_f64(v_reinterpret_as_s64(u1)));
+        }
+        s += v_reduce_sum(v_add(r0, r1));
+        for (; j < n; j++) {
+            double v = saturate_cast<double>(src[j]);
+            s += v * v;
+        }
+        return s;
+    }
+};
+
+template<>
+struct NormL2_SIMD<short, double> {
+    double operator() (const short* src, int n) const {
+        int j = 0;
+        double s = 0.f;
+        v_float64 r0 = vx_setzero_f64(), r1 = vx_setzero_f64();
+        for (; j <= n - 2 * VTraits<v_int16>::vlanes(); j += 2 * VTraits<v_int16>::vlanes()) {
+            v_int16 v0 = vx_load(src + j);
+            r0 = v_add(r0, v_cvt_f64(v_dotprod_expand(v0, v0)));
+
+            v_int16 v1 = vx_load(src + j + VTraits<v_int16>::vlanes());
+            r1 = v_add(r1, v_cvt_f64(v_dotprod_expand(v1, v1)));
+        }
+        s += v_reduce_sum(v_add(r0, r1));
+        for (; j < n; j++) {
+            double v = saturate_cast<double>(src[j]);
+            s += v * v;
+        }
+        return s;
+    }
+};
+
+template<>
+struct NormL2_SIMD<int, double> {
+    double operator() (const int* src, int n) const {
+        int j = 0;
+        double s = 0.f;
+        v_float64 r0 = vx_setzero_f64(), r1 = vx_setzero_f64();
+        for (; j <= n - 2 * VTraits<v_int32>::vlanes(); j += 2 * VTraits<v_int32>::vlanes()) {
+            v_int32 v0 = vx_load(src + j);
+            r0 = v_add(r0, v_cvt_f64(v_dotprod(v0, v0)));
+
+            v_int32 v1 = vx_load(src + j + VTraits<v_int32>::vlanes());
+            r1 = v_add(r1, v_cvt_f64(v_dotprod(v1, v1)));
+        }
+        s += v_reduce_sum(v_add(r0, r1));
+        for (; j < n; j++) {
+            double v = src[j];
+            s += v * v;
+        }
+        return s;
+    }
+};
+
+template<>
+struct NormL2_SIMD<float, double> {
+    double operator() (const float* src, int n) const {
+        int j = 0;
+        double s = 0.f;
+        v_float64 r00 = vx_setzero_f64(), r01 = vx_setzero_f64();
+        v_float64 r10 = vx_setzero_f64(), r11 = vx_setzero_f64();
+        for (; j <= n - 2 * VTraits<v_float32>::vlanes(); j += 2 * VTraits<v_float32>::vlanes()) {
+            v_float32 v0 = vx_load(src + j), v1 = vx_load(src + j + VTraits<v_float32>::vlanes());
+            v_float64 v00 = v_cvt_f64(v0), v01 = v_cvt_f64_high(v0);
+            v_float64 v10 = v_cvt_f64(v1), v11 = v_cvt_f64_high(v1);
+            r00 = v_fma(v00, v00, r00); r01 = v_fma(v01, v01, r01);
+            r10 = v_fma(v10, v10, r10); r11 = v_fma(v11, v11, r11);
+        }
+        s += v_reduce_sum(v_add(v_add(v_add(r00, r01), r10), r11));
+        for (; j < n; j++) {
+            double v = src[j];
+            s += v * v;
+        }
+        return s;
+    }
+};
+
+template<>
+struct NormL2_SIMD<double, double> {
+    double operator() (const double* src, int n) const {
+        int j = 0;
+        double s = 0.f;
+#ifndef CV_RVV // It triggers segmentation fault on CI but it works fine on real hardware
```

Comment:
This is introduce to workaround failed tests on ci of rvv node.

The RVV CI node is getting segmentation fault on the following line for some reason

https://github.com/opencv/opencv/blob/b44b30b7303f4beb8e71d59673f9a019045c0a95/modules/core/test/test_mat.cpp#L827

But it works fine on real hardware.